ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign.tsx

```diff
@@ -24,7 +24,9 @@ const TypedSignInfo: React.FC = () => {
     return null;
   }
 
-  const { domain = {} } = JSON.parse(currentConfirmation?.msgParams?.data);
+  const { domain = {} } = JSON.parse(
+    currentConfirmation?.msgParams?.data as string,
```

Comment:
No need for optional chainig.. we have arleady validated in the previous if clause

Reply:
Also would love if we could avoid type casting

Reply:
Yep optional chaining can be avoided here but typecasting not